### PR TITLE
Release/v1.0.0

### PR DESCRIPTION
**Pull request by ullrichd21**
# Release v1.0.0

#### New Features:

- GatorYAML removed locally, replaced by GatorYAML dependency
- Added default assignment name functionality
- Added more detailed help text
- Added a dropdown for GatorGrader and GatorGradle in the GUI
- Changed format for default checks
- Updated from PyQt5 -> PyQt6
- GUI now available as an **optional** install!

#### Bug Fixes / Minor Changes:
- Various extraneous files removed
- Fixed destructive file creation
- Fixed errors in help text
- Fixed styling issues
- Fixed PyQt6 test failures
- Fixed various issues in test cases

### Type of Change

- [x] Feature
- [x] Bug fix
- [ ] Documentation

## Contributors

- @ullrichd21 
- @WesleyL30 
- @ColemanKobe 
- @PaigeCD 
- @favourojo 

## Reminder

All GitHub Actions should be in a passing state before any pull request is merged.

### CODE REVIEW

File: gatorconfig/gator_config.py [hunk 1]
```diff
@@ -2,54 +2,101 @@
...
+            "break": brk,
+            "fastfail": fastfail,
+            "readme": gen_readme,
+            "indent": indent,
+            "commits": commit_count,
+        }
+        file_yaml = gatoryaml.dump(header, body)
+        output_file(file_yaml, output_path)
+    elif config_path.exists():
+        print(f"'gatorgrader.yml' already exists within {config_dir}")
```
Comment (Michionlion) [inline gatorconfig/gator_config.py:56]: This entire block is duplicated below (starting on line 58), so your main generation runs twice. In fact, when running your tool this can be seen in the fact that it will *always* complain about `gatorgrader.yml` already existing (because it was created in this first run, and so the second finds that it already exists).

File: tests/test_gator_config.py [hunk 2]
```diff
@@ -26,34 +27,42 @@ def test_cli_input(mocker, tmpdir):
...
     with open(test_file, encoding='utf-8') as fle:
-        #print(fle.read())
-        assert "name: Project" in fle.read()
-
+        assert "name: GatorConfig" in fle.read()
 
+def test_cli_overwrite(tmpdir):
+    """Test cli \"--overwrite\" flag"""
+    test_dir = tmpdir.mkdir("testing")
+    test_file = test_dir / "config" / "gatorgrader.yml"
```
Comment (mariakimheinert) [inline tests/test_gator_config.py:52]: If testing the overwriting function, a file at this path should be created before running the program.

File: gatorconfig/gator_config.py [hunk 3]
```diff
@@ -2,54 +2,101 @@
...
-    brk: bool = typer.Option(False, "--break"),
-    fastfail: bool = typer.Option(False),
-    gen_readme: bool = typer.Option(False),
-    file: List[str] = typer.Option([]),
-    #language: str = typer.Option(None),
-    output_path: Path = typer.Option(Path.cwd()),
-    indent: int = typer.Option(4),
-    commit_count: int = typer.Option(5)
+    name: str = typer.Option(default_name(), help="The name of the project"),
+    brk: bool = typer.Option(True, "--break", help="Enables break"),
```
Comment (Michionlion) [inline gatorconfig/gator_config.py:25]: Instead of making this option enable break, it should be called `--no-break` and set the `break` value in the configuration to `false` if it is specified (i.e. if it is true); as it is now, there is no way for a user to create a configuration file with `break: false` in it. This is pretty important to the functionality of your program, so let me know if I can clarify this.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Excellent work on this final release! I have one small requested fix.

File: gatorconfig/gator_config.py [hunk 4]
```diff
@@ -77,10 +124,12 @@ def get_checks(file: List[Path]) -> Dict:
...
         running = True
-        print("")
-        check_list = []
+        check_list = ['--description "Confirm the file exists" ConfirmFileExists',
+                      '--description "Make sure there are no TODOs in the file"'
+                      ' MatchFileFragment --fragment "TODO" --count 0']
+        print(f"These checks are added by default:\n {check_list} \n")
         while running:
-            check = input(f"Enter a check for {item} (Press \"Enter\" to move on): ")
+            check = input(f"Enter a check for {item} (Press 'Enter' to move on): ")
```
Comment (Michionlion) [inline gatorconfig/gator_config.py:132]: This should probably say "to move to the next file", as "move on" is ambiguous.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
One more requested change--please update your `README.md` so that it contains all required parts of project documentation: https://mariakimheinert.com/software-innovation-2/assessment-strategy/final-release/#project-documentation

**Review by Michionlion — CHANGES_REQUESTED** [review]
Great work on these improvements! There are a few bugs I've found, along with some code improvements; I've listed the feedback that isn't line-specific here, and you can read the rest of the feedback below.

1) Your README.md will become the displayed front page of your tool on PyPI, so it should be user-facing and include details on how to install the tool with `pip` or `pipx`.

2) Creating a GitHub Action will fail with an error if `.github/workflows` does not exist. You should automatically create that directory, if it doesn't exist. :eyes: 

3) Your implementation currently writes `commits: <value>` to the header of `gatorgrader.yml`, but that is invalid configuration -- there is no `commits` parameter for GatorGradle. I mentioned this below as well, but you should think about removing this option all-together.

4) In the GUI, you cannot select a file (only folders) in Reflection's **Open** button-triggered file chooser.

5) When using the GUI, any checks added end up using the full absolute path to the file in `gatorgrader.yml`, instead of a path relative to the root of the assignment. :eyes: 

6) When a file is in the root directory of the assignment and is passed as an argument to `--file`, an empty `:` is added in the `gatorgrader.yml` output, like: :eyes: 

```
README.md:
    :
        --description "Confirm the file exists" ConfirmFileExists
```

Overall, this tool looks extremely useful and full-featured, and although there are a few bugs that are affect the user experience pretty heavily, they should be relatively straightforward to fix. I'm looking forward to reviewing your final release!

**Note**: if you're running low on time, the critical issues that really should be fixed for your release (versus shelving them as an issue) are marked with :eyes:.

File: gatorconfig/gui/form.py [hunk 5]
```diff
@@ -131,18 +169,28 @@ def submit_form(self):
...
-                     "files": files
+        full_data = {
+                        "header": {
+                            "name": self.assignment_name.text(),
+                            "break": self.break_fail.isChecked(),
+                            "fastfail": self.fast_fail.isChecked(),
+                            "indent": int(self.indent_size.text()),
+                            "idcommand": "",
+                            "version": self.get_grader_version(),
+                            "executables": "",
```
[the remark this answers is not in the document]
Comment (Michionlion) [reply gatorconfig/gui/form.py:180]: This results in `idcommand:` in the header of `gatorgrader.yml`, which is invalid configuration.

**Review by Michionlion — COMMENTED** [review]
I missed this before, but the GUI is asking for a GatorGradle version; however, your tool does not output a `build.gradle` file (which is what would contain that version). If you've completed everything else, that may be a feature that could be added. If not, that is definitely future work.

File: pyproject.toml [hunk 6]
```diff
@@ -4,16 +4,21 @@ version = "0.1.0"
 description = "Autogeneration of GatorGradle configuration files."
 authors = ["Daniel Ullrich", "Kobe Coleman", "Paige Downey", "Favour Ojo", "Wesley Long"]
 license = "MIT"
+readme = "README.md"
+homepage = "https://github.com/cmpsc-481-s22-m1/GatorConfig"
+repository = "https://github.com/cmpsc-481-s22-m1/GatorConfig"
```
[the remark this answers is not in the document]
Comment (PaigeCD) [reply pyproject.toml:9]: Are there any classifiers in particular that you would recommend for this tool?

[the remark this answers is not in the document]
Comment (ColemanKobe) [reply pyproject.toml:9]: Hopefully, this link can provide some ideas: https://pypi.org/classifiers/

File: gatorconfig/gator_config.py [hunk 7]
```diff
@@ -65,51 +51,50 @@ def cli_input(
             # Creation of the output variable
             header = {
                 "name": name,
-                "break": brk,
+                "break": not no_break,
                 "fastfail": fastfail,
+                "readme": gen_readme,
```
Comment (Michionlion) [inline gatorconfig/gator_config.py:56]: This ends up generating `readme: False/True` in `gatorgrader.yml`, which is not valid configuration.

File: gatorconfig/gui/form.py [hunk 8]
```diff
@@ -175,15 +169,15 @@ def submit_form(self):
                             "break": self.break_fail.isChecked(),
                             "fastfail": self.fast_fail.isChecked(),
                             "indent": int(self.indent_size.text()),
-                            "idcommand": "",
                             "version": self.get_grader_version(),
-                            "executables": "",
                             "startup": self.startup_script_text.text(),
```
Comment (Michionlion) [inline gatorconfig/gui/form.py:173]: `startup` should be handled like `idcommand` and the others, since there is no default value.
Additionally, it would be nice if `break` defaulted to checked.

File: gatorconfig/gui/form.py [hunk 9]
```diff
@@ -175,15 +169,15 @@ def submit_form(self):
...
-                            "reflection": ""
                         },
                         "body": files
                      }
 
+        full_data = self.insert_idcommand(full_data)
+        full_data = self.insert_executables(full_data)
+
```
Comment (Michionlion) [inline gatorconfig/gui/form.py:180]:
```suggestion
full_data = self.insert_reflection(full_data)
```

Comment (Michionlion) [reply gatorconfig/gui/form.py:180]: Seems like you forgot to call this function.

**Review by Michionlion — CHANGES_REQUESTED** [review]
Great work everyone! Just a few small, quick changes. Additionally, although it doesn't affect your build in any way, it would be nice to update the version in `pyproject.toml` to be current.

File: gatorconfig/gator_config.py (hunk 7 above)
Comment (ColemanKobe) [reply gatorconfig/gator_config.py:55]: For me, it doesn't do that, at least with this PR I just merged

File: gatorconfig/gui/form.py [hunk 10]
```diff
@@ -189,23 +190,19 @@ def get_grader_version(self):
             return self.grader_version.currentText()
         return self.grader_version.text()
 
-    def insert_idcommand(self, data):
-        """Returns full_data header with added idcommand if the user specified one."""
-        txt = self.idcommand.text()
-        if txt != "":
-            data["header"]["idcommand"] = txt
-        return data
+    def insert_checked_commands(self, data):
```
Comment (Michionlion) [inline gatorconfig/gui/form.py:193]: This new method (which is a better way to do this than what you had before) is not called.

Comment (Michionlion) [reply gatorconfig/gui/form.py:193]: Although definitely not something you should fix in this release, it would be nice to have some kind of way to test these things in (at least) an integration test for the GUI.

**Review by Michionlion — CHANGES_REQUESTED** [review]
Looks good -- minor GUI issue though! After that's fixed you should be good to go.

**Review by Michionlion — DISMISSED** [review]
Looks good!

File: gatorconfig/gui/cwidgets.py [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+        hbox.addWidget(self.button)
+        self.setLayout(hbox)
+
+    def set_textbox_text(self, text):
+        """Set the textbox text"""
+        self.line.setText(os.path.relpath(text))
+
+    def get_file_path(self):
+        """Return the file path from the text box"""
+        return self.line.text()
```
Comment (mariakimheinert) [inline gatorconfig/gui/cwidgets.py:35]: Currently, running the GUI produces this error:

```
ykim@aldenv8362 GatorConfig % poetry run gatorconfig --overwrite --gui
Traceback (most recent call last):
  File "/Users/ykim/CoursePrep/s22m1/481/project-testing/GatorConfig/gatorconfig/gui/homepage.py", line 31, in <lambda>
    submit_button = QPushButton("Submit", clicked=lambda: self.submit_form(form))
  File "/Users/ykim/CoursePrep/s22m1/481/project-testing/GatorConfig/gatorconfig/gui/homepage.py", line 48, in submit_form
    data = form.submit_form()
  File "/Users/ykim/CoursePrep/s22m1/481/project-testing/GatorConfig/gatorconfig/gui/form.py", line 178, in submit_form
    full_data = self.insert_checked_commands(full_data)
  File "/Users/ykim/CoursePrep/s22m1/481/project-testing/GatorConfig/gatorconfig/gui/form.py", line 199, in insert_checked_commands
    if self.reflection.text() != "":
AttributeError: 'CFilePicker' object has no attribute 'text'
zsh: abort      poetry run gatorconfig --overwrite --gui
ykim@aldenv8362 GatorConfig % poetry run gatorconfig --overwrite --gui
Traceback (most recent call last):
  File "/Users/ykim/CoursePrep/s22m1/481/project-testing/GatorConfig/gatorconfig/gui/homepage.py", line 31, in <lambda>
    submit_button = QPushButton("Submit", clicked=lambda: self.submit_form(form))
  File "/Users/ykim/CoursePrep/s22m1/481/project-testing/GatorConfig/gatorconfig/gui/homepage.py", line 48, in submit_form
    data = form.submit_form()
  File "/Users/ykim/CoursePrep/s22m1/481/project-testing/GatorConfig/gatorconfig/gui/form.py", line 178, in submit_form
    full_data = self.insert_checked_commands(full_data)
  File "/Users/ykim/CoursePrep/s22m1/481/project-testing/GatorConfig/gatorconfig/gui/form.py", line 200, in insert_checked_commands
    data["header"]["reflection"] = self.reflection.text()
AttributeError: 'CFilePicker' object has no attribute 'text'
zsh: abort      poetry run gatorconfig --overwrite --gui
```

To resolve it, you can just change the name of this function. Additionally, the path should be relative, so `os.path.relpath` should be used.
```suggestion
    def text(self):
        """Return the file path from the text box"""
        if self.line.text() == "":
            return ""
        return os.path.relpath(self.line.text())
```
(The check for an empty string is because `relpath` will not accept empty strings.)

**Review by mariakimheinert — CHANGES_REQUESTED** [review]

**Review by mariakimheinert — APPROVED** [review]
Awesome work, y'all. Thanks for all your hard work!

**Review by Michionlion — APPROVED** [review]
I think there might be a few other edge cases in the GUI (it can be a bit complicated to test!), but this should be good to go, and extremely useful for professors! Thank you for all your hard work, everyone!
